packages/frontend/src/components/fields-editor/FieldsEditor.tsx

```diff
@@ -0,0 +1,215 @@
+import Card from "@mui/material/Card";
+import CardContent from "@mui/material/CardContent";
+import {
+  Button,
+  FormControl,
+  Grid2,
+  InputLabel,
+  MenuItem,
+  Select,
+  TextField,
+  Typography,
+} from "@mui/material";
+import { EditableBridgeConfig } from "../bridge/BridgeConfigEditor";
+import {
+  HomeAssistantMatcher,
+  HomeAssistantMatcherType,
+} from "@home-assistant-matter-hub/common";
+import Divider from "@mui/material/Divider";
+import AddCircleOutlineIcon from "@mui/icons-material/AddCircleOutline";
+import DeleteOutlineIcon from "@mui/icons-material/DeleteOutline";
+import { useEffect, useState } from "react";
+
+interface FieldsEditorProps {
+  value: EditableBridgeConfig;
+  onChange: (event: { value: EditableBridgeConfig; isValid: boolean }) => void;
+}
+
+const FieldsEditor = ({ value, onChange }: FieldsEditorProps) => {
+  const [fieldValues, setFieldsValues] = useState<EditableBridgeConfig>(
+    () => value,
+  );
+
+  const handleOnChange = (
+    newIncludesValue: HomeAssistantMatcher[],
+    key: "include" | "exclude",
+  ) => {
+    const data = JSON.parse(JSON.stringify(value));
+    data.filter[key] = newIncludesValue;
+    setFieldsValues(data);
+  };
+
+  useEffect(() => {
+    if (JSON.stringify(value) === JSON.stringify(fieldValues)) return;
+
+    const includesFinted = fieldValues.filter.include.find(
+      (e) => !e.type || !e.value,
+    );
+    const excludeFinted = fieldValues.filter.exclude.find(
+      (e) => !e.type || !e.value,
+    );
+    if (includesFinted || excludeFinted) {
+      onChange({ isValid: false, value: fieldValues });
+      return;
+    }
+    onChange({ isValid: true, value: fieldValues });
+  }, [fieldValues, value, onChange]);
```

Comment:
i'd call `onChange` in `handleOnChange` and would remove that whole `useEffect` here.